### PR TITLE
Home page add buy report sellers page functionality

### PR DESCRIPTION
**Pull request by hripman**

### CODE REVIEW

File: src/pages/home/components/product/Report.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        this.props.handleOnSubmit({
+            option: this.state.checked,
+            comment: get(this.refs, 'reportComment.value')
+        });
+    }
+
+    render() {
+        let reportNoteField = null;
+
+        if (this.state.checked === '3') {
```
Comment (shahen94) [inline src/pages/home/components/product/Report.js]: Why `'3'` should be string ? It's little bit confusing.
Also you can use constant to make this code more readable .
For example.

```
const reportTypes = {
     COPYRIGHT: 1,
     NUDITY: 2,
     PROFANITY: 3,
     RACISM: 4,
     SPAM: 5,
     IRRELEVANT: 6
};

this.setState({ checked: reportTypes.RACISM });
```

File: src/pages/home/components/seller/SellerContainer.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    render() {
+        const {requestState, seller} = this.props;
+
+        if (requestState.pending) {
+            return <Spinner name='circle'/>
+        }
+
+        let sellerTemplate = null;
+        if (seller) {
+            sellerTemplate = seller.map((o, index) => <div className="col-xs-3" key={index}><Seller
```
Comment (shahen94) [inline src/pages/home/components/seller/SellerContainer.js]: use index as `key` is not a good idea and may produce some performance issues in the future.

Please try to use another value for `key` prop.

File: src/pages/home/components/seller/SellerContainer.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    }
+}
+
+const mapStateToProps = (state) => {
+    return {
+        requestState: state.requestState,
+        seller: state.seller
+    }
+}
+
```
Comment (gnunua) [inline src/pages/home/components/seller/SellerContainer.js]: Let's use E6 object destructuring and Shorthand property names
```
const mapStateToProps = ({requestState, seller}) => {
     return {
         requestState,
         seller,
     }
 }

```

Comment (gnunua) [reply src/pages/home/components/seller/SellerContainer.js]: Maybe you would like to prefer  more short way writing `mapStateToProps` :
```
 const mapStateToProps = ({requestState, seller}) => ({
         requestState,
         seller,
 })
```

File: src/pages/home/components/product/ProductContainer.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+const mapStateToProps = (state) => {
+    return {
+        requestState: state.requestState,
+        products: state.product
+    }
+}
+
+const mapDispatchToProps = (dispatch) => {
```
Comment (gnunua) [inline src/pages/home/components/product/ProductContainer.js]: Redux has `bindActionCreators`. Check [here](https://redux.js.org/docs/api/bindActionCreators.html). I think `bindActionCreators` `mapDispatchToProps` method is more clear.

**Pull request merged**